Upgrade D3 6.3.1

iezer opened the pull request:
Upgrade D3 to 6.3.1. See the [migration guide](https://observablehq.com/@d3/d3v6-migration-guide).

Changes
 * Make all D3 imports in app (not test) code from specific modules rather than `d3`. 
 * D3.event was removed and now the event object is passed to handlers. Changed `ZoomContainer` to reflect this.
 * Fix Histogram test
 * Confirm all `d3-array` dependencies in `package-lock.json` are pointing to `2.9.1` and not `1.x.x`.

We will release this as a major version bump in Reactochart. 4.0!

ekh64 reviewed (commented):
I'm all for using the modules, just curious if there's any performance improvements to using them (eg tree shaking, reduced bundle size)

I wonder if it might be good to introduce a lint rule for using d3 modules over d3: https://eslint.org/docs/rules/no-restricted-imports

ekh64 commented:
Also I wonder if now is a good time to run prettier over the entire codebase, I saw some formatting changes in some of your files. Maybe there are others that could use a format, too.

iezer commented:
@ekh64 Thanks for taking a look! To answer your questions:
 * I ran prettier on all the files. It was mostly tests that still had failures and now the automated diffs are fixed.
 * I removed the global d3 imports from all the test files. Not sure about the build size impact because we don't produce a single payload as part of our build process. It's up to the consumer to build so it's depends on the consumer's webpack config. I did not that the app was depending on multiple versions of `d3-array` and now it just requires 1.
 * I've added the `es-lint` check from importing `d3`. Thanks for the suggestion

ekh64 reviewed (approved):
look great!
